### PR TITLE
doc: Fixup getrawtransaction RPC docs

### PR DESCRIPTION
**Pull request by dougEfresh**
Added  `ScriptPubKeyDoc` function

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Thanks,

ACK 97115de1839c69d63cae6ea772c17905739bf0ae

**Review by kristapsk — APPROVED** [review]
cr utACK 97115de1839c69d63cae6ea772c17905739bf0ae

**Comment by MarcoFalke** [comment]
Rendered diff:


```diff
diff --git a/decoderawtransaction b/decoderawtransaction
index 05a49fe..7600def 100644
--- a/decoderawtransaction
+++ b/decoderawtransaction
@@ -45,8 +45,8 @@ Result:
         "asm" : "str",        (string) Disassembly of the public key script
         "desc" : "str",       (string) Inferred descriptor for the output
         "hex" : "hex",        (string) The raw public key script bytes, hex-encoded
-        "type" : "str",       (string) The type, eg 'pubkeyhash'
-        "address" : "str"     (string, optional) The Bitcoin address (only if a well-defined address exists)
+        "address" : "str",    (string, optional) The Bitcoin address (only if a well-defined address exists)
+        "type" : "str"        (string) The type (one of: nonstandard, pubkey, pubkeyhash, scripthash, multisig, nulldata, witness_v0_scripthash, witness_v0_keyhash, witness_v1_taproot, witness_unknown)
       }
     },
     ...
diff --git a/getrawtransaction b/getrawtransaction
index 462f051..6750205 100644
--- a/getrawtransaction
+++ b/getrawtransaction
@@ -8,8 +8,8 @@ the specified block is available and the transaction is in that block.
 Hint: Use gettransaction for wallet transactions.
 
 If verbosity is 0 or omitted, returns the serialized transaction as a hex-encoded string.
-If verbosity is 1, returns a JSON Object with information about transaction.
-If verbosity is 2, returns a JSON Object with information about transaction, including fee and prevout information.
+If verbosity is 1, returns a JSON Object with information about the transaction.
+If verbosity is 2, returns a JSON Object with information about the transaction, including fee and prevout information.
 
 Arguments:
 1. txid         (string, required) The transaction id
@@ -59,8 +59,8 @@ Result (if verbosity is set to 1):
         "asm" : "str",               (string) Disassembly of the public key script
         "desc" : "str",              (string) Inferred descriptor for the output
         "hex" : "hex",               (string) The raw public key script bytes, hex-encoded
-        "type" : "str",              (string) The type, eg 'pubkeyhash'
-        "address" : "str"            (string, optional) The Bitcoin address (only if a well-defined address exists)
+        "address" : "str",           (string, optional) The Bitcoin address (only if a well-defined address exists)
+        "type" : "str"               (string) The type (one of: nonstandard, pubkey, pubkeyhash, scripthash, multisig, nulldata, witness_v0_scripthash, witness_v0_keyhash, witness_v1_taproot, witness_unknown)
       }
     },
     ...
@@ -72,9 +72,9 @@ Result (for verbosity = 2):
   ...,                               Same output as verbosity = 1
   "fee" : n,                         (numeric, optional) transaction fee in BTC, omitted if block undo data is not available
   "vin" : [                          (json array)
-    {                                (json object, optional) utxo being spent, omitted if block undo data is not available
+    {                                (json object) utxo being spent, omitted if block undo data is not available
       ...,                           Same output as verbosity = 1
-      "prevout" : {                  (json object) Only if undo information is available)
+      "prevout" : {                  (json object, optional) Only if undo information is available)
         "generated" : true|false,    (boolean) Coinbase or not
         "height" : n,                (numeric) The height of the prevout
         "value" : n,                 (numeric) The value in BTC

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -206,25 +209,18 @@ static RPCHelpMan getrawtransaction()
                         RPCResult::Type::OBJ, "", "",
                         {
                             {RPCResult::Type::ELISION, "", "Same output as verbosity = 1"},
-                            {RPCResult::Type::NUM, "fee", /* optional */ true, "transaction fee in " + CURRENCY_UNIT + ", omitted if block undo data is not available"},
+                            {RPCResult::Type::NUM, "fee", /*optional=*/true, "transaction fee in " + CURRENCY_UNIT + ", omitted if block undo data is not available"},
                             {RPCResult::Type::ARR, "vin", "",
                             {
-                                {RPCResult::Type::OBJ, "", /* optional */ true, "utxo being spent, omitted if block undo data is not available",
+                                {RPCResult::Type::OBJ, "", "utxo being spent, omitted if block undo data is not available",
```
Comment (MarcoFalke) [inline src/rpc/rawtransaction.cpp:215]: Sorry for missing this earlier, but I think the ", omitted ..." part will need to be removed as well?

Comment (dougEfresh) [reply src/rpc/rawtransaction.cpp:215]: @MarcoFalke see https://github.com/bitcoin/bitcoin/pull/26968

**Pull request merged**